### DISCUSSION TITLE
Environmental impact

### DISCUSSION THREAD

**Discussion by husseinalhammad** [email-markup-consortium/email-markup-consortium#39] [General]
I believe documenting things is a key in making progress here. One thing I think about every now and then is what the environmental impact I have as a developer. In 2022, it's naive to simply think digital = always better for the environment.

When I think of what an email developer has to do to test emails:

1. Upload HTML to a testing tool (e.g. Testi@, Litmus, Email on Acid)
2. The testing tool sends the email to a number of email accounts. Not every email client requires a unique email account, but some do.
3. The testing tool would probably have a testing lab with several devices automated to take screenshots of received emails
4. The testing tool uploads screenshots from multiple devices to a cloud storage (stored indefinitely in a data centre?)
5. Email service providers receive, processe and store the HTML email (stored indefinitely in a data center?)

Is there a significant enironmental impact here? Would acheiving our goals decrease this impact in any way? 

Even a primitive approach to testing emails where the developer manually sends the email, an email service provider still need to store a test email (see 5 above).

**Comment by Malvoz** [comment]
Just noting that the W3C <span title="Technical Architecture Group">TAG</span>'s Ethical Web Principle: [The web must be an environmentally sustainable platform](https://w3ctag.github.io/ethical-web-principles/#sustainable) may be used as a reference, where appropriate.

**Reply by husseinalhammad** [reply]
This is a really good one. Thanks, @Malvoz

**Comment by M-J-Robbins** [comment]
Yes I agree, reducing the need for such wide testing would help. 

Also improving the interoperability of the markup could dramatically reduce file sizes, as fewer workarounds and fallbacks would be needed. That would lead to less storage in the ESP's and Email Clients and less data being transferred.  

But would senders just add more content?  Should we be looking at a more formalized hard limit? Currently, senders aim to keep HTML email files under ~100kb because of [Gmail clipping limit](https://github.com/hteumeuleu/email-bugs/issues/41), and potential [deliverability issues of large emails](https://www.emailonacid.com/blog/article/email-development/how_does_email_file_size_affect_deliverability/).  Is that enough? If Gmail remove or adjust that clipping limit would senders start sending bigger HTML files?

Also on the subject of environmental impact, there is an interesting idea going around about email expiration at [zerocarbon.email](https://www.zerocarbon.email). The most important thing here is the recipient needs to be in control of if a message is deleted or not.

**Reply by husseinalhammad** [reply]
I was thinking too much about testing and forgot live emails 😅

## Less code != less content
Reducing the file size (at least reducing the code) would be a plus. Not having to inline CSS would certainly help here. 

As for content, I find some newsletters to be content heavy, but light in terms of code. And with the increase of exclusive content and paid newsletters, there's definitely an audience who want a lot of content in email. I don't think that's wrong even though I personally don't like consuming content in my inbox. 

More content can be ok. More unnecessary code is problematic to everyone involved.


## Email expiration
Interesting. As a developer, I'd most likely set an expiry date on email tests. I can also see this being used for a certain type of emails (password resets, magic login links, etc.).

**Reply by husseinalhammad** [reply]
Looking at [zerocarbon.email](https://www.zerocarbon.email/) again, it seems there's an overlap here. I wonder if we can collaborate in documenting the environmental impact, which can potentially help both parties.

**Reply by M-J-Robbins** [reply]
Perhaps this can add to the [Email Platform Status](https://github.com/email-markup-consortium/email-markup-consortium/discussions/41#discussioncomment-2255570) task and gather email file sizes and content size. Maybe some kind of code to content ratio, using X characters of code to generate Y characters of content.

**Reply by jloriaux** [reply]
Really glad to see environmental impact to be a topic here ;-)

We recently worked with a specialist of the Digital Environmental Impact (we have webinar on Youtube next week with him, but in Frenchhttps://www.youtube.com/watch?v=r-wX_xOtr8Y) about what are the steps that produce the bigger part of carbon emissions when we talk about email marketing.

And the answer is clear, it's the lifecycle of the device (production, usage, recycling) of the recipient who is responsible for the bigger part of carbon emissions. The first objective would therefore be to increase the life duration of the devices. To achieve this, the devices must not become obsolete (new incompatible software, too heavy, ...)

With email this is not a big problem. In general we stay on standards that do not create obsolescence. But if we see big technological breakthroughs in email, it should not force (or add a reason) to change the device.

About Email Expiration Dates, we excluded to work with expiry date stored in the content. Using smtp headers allows us to use a technology that is mastered by all MBPs. Parsing the headers also requires less energy. The Draft RFC is here : https://datatracker.ietf.org/doc/html/draft-billon-expires

Happy to continue this discussion ;-)

Jonathan

**Reply by husseinalhammad** [reply]
Thanks for your input, @jloriaux. This is certainly helpful.

I wonder whether there is more to investigate here beyond the storage of emails the end user receives. For example, the testing cycle of a HTML email that many email developers go through on a daily basis makes use of testing tools as outlined in the main post (send emails to multiple inboxes, have automated tasks to take screenshots of these sent emails, etc.). These testing tools are amazing, but perhaps overused out of necessity (there is no alternative testing method and email clients do not handle HTML in the same way).

**Reply by jloriaux** [reply]
@husseinalhammad Maybe the right time to ask Litmus, Email on Acid and co how they handle ecodesign of their solutions, where they host and if they are doing carbon accounting of their activities.

**Reply by husseinalhammad** [reply]
@jloriaux Yes, agreed. We'll attempt to talk to those closer to the testing solutions. Thanks again!

**Comment by alicemeichi** [comment]
I'm not entirely sure that the concerns around the personal devices on which emails are stored comprise the entire equation here. They may for emails delivered via POP3. But with IMAP or Webmail where emails are stored on a server when they hit an inbox, the fileweights of individual emails will have an impact on an environment. 

For example: A marketer sending out a 100kb email to a list of 10,500 webmail recipients is generating 1GB of new data that will have to be stored somewhere, which impacts the carbon footprints of datacenters. Cutting that 100kb email down rather than cutting marketing mailing lists down strikes me as more attractive way for marketers to have a positive environmental impact.

This article below even goes into how not sending out email attachments is a way to reduce carbon footprints. Email attachments, like embedded images or markup fileweight, are similarly stored on email servers with a similar impact on carbon footprints.

https://medium.datadriveninvestor.com/save-the-planet-replace-email-attachments-with-file-share-links-1f643e06c0d3

**Reply by alicemeichi** [reply]
Another source: https://www.greenmatters.com/p/do-emails-leave-carbon-footprint

> Most emails, including spam ones, unread ones, and those you simply forgot you never deleted are generally stored on the cloud. Cloud storage requires quite a lot of electricity, which in most parts of the world is still generated by fossil fuels.
>
>[...] 
>
> As calculated by [Eco2 Greetings](https://www.eco2greetings.com/News/The-Carbon-Footprint-of-Email-vs-Postal-Mail.html#:~:text=A%20normal%20email%2C%20one%20without,filtering%20and%20reading%20your%20messages.&text=Every%20email%20processed%20uses%20electricity.), text-based emails emit about 4 grams of CO2e (carbon dioxide equivalent). The source also estimates that an average year of emailing emits about 136 kilograms of CO2e, which is about the same impact as driving 200 miles in a gas-powered car.

**Comment by M-J-Robbins** [comment]
Another company focusing on environmental impact in email marketing https://ecosend.io

**Comment by sarawillimon** [comment]
I will reach out to DotDigital, who has gone carbon neutral and see if they would be interested in partnering with us or providing any resources on how to best measure impact of email marketing & how EMC's work can influence environmental impacts.

**Reply by husseinalhammad** [reply]
That would be super helpful. Thanks, @sarawillimon

**Comment by alicemeichi** [comment]
Another thought that occurred to me is that many tech companies are incentivized to reduce their Cloud consumption (and therefore environmental impact) because they're charged for usage. 

But unfortunately the financial cost of environmentally-unfriendly emails usually lands on the recipient as their inbox storage fills up, so marketers are less incentivized to keep their markup fileweight down because the user absorbs these costs.

At the moment, many ESPs charge per email sent... but what if they were to charge per KB/MB/GB sent instead? That could be an angle where we could approach ESPs to green their pricing structure.

**Reply by husseinalhammad** [reply]
If the sender needs to process the HTML before sending, could the size of the HTML impact how much processing they do? Not sure how significant this would be. It may depend on the type of processing they do (updating URLs for click tracking, CSS inlining, etc.).

**Reply by sarawillimon** [reply]
Those are great thoughts and questions. Another piece that would be interesting to compare, on a related note, is the impact of the use of various WYSIWYGs, which typically add a lot of coding weight, vs custom coded. 
Is there opportunity for best practices to keep wysiwyg functions from bulking up code too much?

**Reply by fernamp** [reply]
These is pretty interesting!
I'm going to try to include this concepts in our next meeting at my job.
(BTW, I just discovered this group now...)
